app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.java

```diff
@@ -1,73 +1,63 @@
 package com.quran.labs.androidquran.database;
 
-import com.quran.labs.androidquran.common.LocalTranslation;
-import com.quran.labs.androidquran.dao.translation.TranslationItem;
-import com.quran.labs.androidquran.util.QuranFileUtils;
-
 import android.content.ContentValues;
 import android.content.Context;
 import android.database.Cursor;
-import android.database.SQLException;
 import android.database.sqlite.SQLiteDatabase;
+import android.support.annotation.NonNull;
 import android.util.SparseArray;
 
+import com.quran.labs.androidquran.common.LocalTranslation;
+import com.quran.labs.androidquran.dao.translation.TranslationItem;
+import com.quran.labs.androidquran.util.QuranFileUtils;
+
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
 import timber.log.Timber;
 
 import static com.quran.labs.androidquran.database.TranslationsDBHelper.TranslationsTable;
 
+@Singleton
 public class TranslationsDBAdapter {
 
-  private SQLiteDatabase mDb;
-  private Context mContext;
-  private static TranslationsDBHelper sDbHelper;
-
-  public TranslationsDBAdapter(Context context) {
-    mContext = context.getApplicationContext();
-    initHelper(mContext);
-  }
-
-  public static synchronized void initHelper(Context context) {
-    if (sDbHelper == null) {
-      sDbHelper = new TranslationsDBHelper(context);
-    }
-  }
+  private final Context context;
+  private final SQLiteDatabase db;
+  private volatile List<LocalTranslation> cachedTranslations;
 
-  public void open() throws SQLException {
-    if (mDb == null && sDbHelper != null) {
-      mDb = sDbHelper.getWritableDatabase();
-    }
+  @Inject
+  public TranslationsDBAdapter(Context context, TranslationsDBHelper adapter) {
+    this.context = context;
+    this.db = adapter.getWritableDatabase();
   }
 
   public SparseArray<LocalTranslation> getTranslationsHash() {
     List<LocalTranslation> items = getTranslations();
 
     SparseArray<LocalTranslation> result = new SparseArray<>();
-    if (items != null) {
-      for (int i = 0, itemsSize = items.size(); i < itemsSize; i++) {
-        LocalTranslation item = items.get(i);
-        result.put(item.id, item);
-      }
+    for (int i = 0, itemsSize = items.size(); i < itemsSize; i++) {
+      LocalTranslation item = items.get(i);
+      result.put(item.id, item);
     }
     return result;
   }
 
+  @NonNull
   public List<LocalTranslation> getTranslations() {
-    if (mDb == null) {
-      open();
-      if (mDb == null) {
-        return null;
-      }
+    List<LocalTranslation> cached = cachedTranslations;
```

Comment:
i'll add a comment - but because the value of `cachedTranslations` can be replaced by a different thread going through `writeTranslationUpdates`, we keep a reference in `cached` (since `cachedTranslations` is volatile, this will be the latest value) and return it (otherwise, between the check and return, the value could change - we want any valid list of local translations, and it's okay if we get a slightly stale one here).